python/ray/serve/_private/deployment_info.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, Dict, Optional
+
+import ray
+from ray.serve._private.autoscaling_policy import BasicAutoscalingPolicy
+from ray.serve._private.common import TargetCapacityDirection
+from ray.serve._private.config import DeploymentConfig, ReplicaConfig
+from ray.serve.generated.serve_pb2 import DeploymentInfo as DeploymentInfoProto
+from ray.serve.generated.serve_pb2 import (
+    TargetCapacityDirection as TargetCapacityDirectionProtoEnum,
+)
+
+# Concurrency group used for operations that cannot be blocked by user code
+# (e.g., health checks and fetching queue length).
+CONTROL_PLANE_CONCURRENCY_GROUP = "control_plane"
+REPLICA_DEFAULT_ACTOR_OPTIONS = {
+    "concurrency_groups": {CONTROL_PLANE_CONCURRENCY_GROUP: 1}
+}
+
+
+TARGET_CAPACITY_DIRECTION_TO_PROTO = {
+    None: TargetCapacityDirectionProtoEnum.UNSET,
+    TargetCapacityDirection.UP: TargetCapacityDirectionProtoEnum.UP,
+    TargetCapacityDirection.DOWN: TargetCapacityDirectionProtoEnum.DOWN,
+}
+
+
+TARGET_CAPACITY_DIRECTION_FROM_PROTO = {
+    TargetCapacityDirectionProtoEnum.UNSET: None,
+    TargetCapacityDirectionProtoEnum.UP: TargetCapacityDirection.UP,
+    TargetCapacityDirectionProtoEnum.DOWN: TargetCapacityDirection.DOWN,
+}
+
+
+class DeploymentInfo:
```

Comment:
I assume this is basically copy-pasted from `common.py` -- is there a reason it needs to be done in this PR?

Reply:
Yes, it's copy-pasted. There was a circular import otherwise. Without the copy-paste:

* `common.py` defined:
    * `DeploymentInfo`, which depends on
        * `BasicAutoscalingPolicy`, which is stored in `autoscaling_policy.py`.
    * `TargetCapacityDirection`
* `autoscaling_policy.py`, which depends on
    * `TargetCapacityDirection`, which is defined in `common.py`.

When `DeploymentInfo` is in `common.py`, `common.py` needs to import `autoscaling_policy.py`. But then `autoscaling_policy.py` needs to import `common.py`.